src/main/java/com/projectkorra/core/util/VectorUtil.java

```diff
@@ -0,0 +1,96 @@
+package com.projectkorra.core.util;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.Location;
+import org.bukkit.util.Vector;
+
+import com.projectkorra.core.math.Angle;
+import com.projectkorra.core.math.Angle.AngleMode;
+
+public class VectorUtil {
+
+	private VectorUtil() {}
+	
+	private static final Vector ZERO = new Vector(0, 0, 0); // this vector should never be modified, only used for equality checks
+	
+	/**
+	 * Get a vector for the direction between two points
+	 * @param from start point
+	 * @param to end point
+	 * @return vector pointing from -> to
+	 */
+	public static Vector direction(Location from, Location to) {
+		return new Vector(to.getX() - from.getX(), to.getY() - from.getY(), to.getZ() - from.getZ());
+	}
+	
+	/**
+	 * Returns an orthogonal vector from an axis
+	 * @param axis vector to be orthogonal of
+	 * @param length magnitude of orthogonal vector
+	 * @param rotation counterclockwise rotation, 0 = righthand side of the axis
+	 * @return an orthogonal vector
+	 */
+	public static Vector orthogonal(Vector axis, double length, Angle rotation) {
```

Comment:
I would prefer that you add the `throws IllegalArgumentException` declaration to the method header that could be caused from the `Validate.isTrue(...)` call. Additionally, it should go in the Javadocs.